connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -91,6 +92,13 @@
     private static final String TRANSFORMS_DOC = "Aliases for the transformations to be applied to records.";
     private static final String TRANSFORMS_DISPLAY = "Transforms";
 
+    public static final String CONFIG_RELOAD_ACTION_CONFIG = "config.action.reload";
+    private static final String CONFIG_RELOAD_ACTION_DOC =
+            "The action to take in order to reload changed configuration values (none or restart).";
```

Comment:
This should probably make more clear that this occurs when configuration values change in external providers, and should describe what none and restart mean. Users shouldn't have to go to the code or KIP to understand this configuration property.